src/tests/ftest/util/run_utils.py

```diff
@@ -536,3 +536,153 @@ def stop_processes(log, hosts, pattern, verbose=True, timeout=60, exclude=None,
     if processes_running:
         log.debug("Processes still running on %s that match: %s", processes_running, pattern_match)
     return processes_detected, processes_running
+
+
+class LocalTask():
+    """Mock ClusterShell.Task to all subprocess.run() to return a RemoteCommandResult."""
+
+    def __init__(self):
+        """Initialize a LocalTask object."""
+        self._host = gethostname().split(".")[0]
+        self._command_output = {}
+        self._return_codes = {}
+        self._timeout_hosts = set()
+
+    def _reset(self):
+        """Reset the command results."""
+        self._command_output = {}
+        self._return_codes = {}
+        self._timeout_hosts.clear()
+
+    def run(self, command, timeout=None):
+        """Run the command locally.
+
+        Args:
+            command (str): command to run.
+            timeout (int, optional): number of seconds to wait for the command to complete.
+                Defaults to None.
+        """
+        self._reset()
+        kwargs = {
+            "encoding": "utf-8",
+            "shell": False,
+            "check": False,
+            "timeout": timeout,
+            "stdout": subprocess.PIPE,
+            "stderr": subprocess.STDOUT,
+        }
+        try:
+            # pylint: disable=subprocess-run-check
+            result = subprocess.run(shlex.split(command), **kwargs)     # nosec
+            self._add_command_output('stdout', result.stdout)
+            self._add_command_output('stderr', result.stderr)
+            self._return_codes[result.returncode] = [self._host]
+
+        except subprocess.TimeoutExpired:
+            # Raised if command times out
+            self._add_command_output('stdout', result.stdout)
+            self._add_command_output('stderr', result.stderr)
+            self._timeout_hosts.add([self._host])
```

Comment:
I expect this would error
```suggestion
            self._timeout_hosts.add(self._host)
```

Reply:
Ah, right.